app/controllers/LegacyApiController.scala

```diff
@@ -30,18 +29,139 @@ object LegacyCreateExplorationalParameters {
     Json.format[LegacyCreateExplorationalParameters]
 }
 
+case class LegacyTaskParameters(taskTypeId: String,
+                                neededExperience: Experience,
+                                pendingInstances: Int,
+                                projectName: String,
+                                scriptId: Option[String],
+                                boundingBox: Option[BoundingBox],
+                                dataSet: String,
+                                datasetId: Option[ObjectId],
+                                editPosition: Vec3Int,
+                                editRotation: Vec3Double,
+                                creationInfo: Option[String],
+                                description: Option[String],
+                                baseAnnotation: Option[BaseAnnotation])
+
+object LegacyTaskParameters {
+  implicit val taskParametersFormat: Format[LegacyTaskParameters] = Json.format[LegacyTaskParameters]
+}
+
 class LegacyApiController @Inject()(annotationController: AnnotationController,
                                     datasetController: DatasetController,
-                                    userController: UserController,
                                     projectController: ProjectController,
-                                    projectDAO: ProjectDAO,
+                                    taskController: TaskController,
                                     organizationDAO: OrganizationDAO,
                                     datasetService: DatasetService,
-                                    taskDAO: TaskDAO,
-                                    taskService: TaskService,
+                                    datasetDAO: DatasetDAO,
                                     sil: Silhouette[WkEnv])(implicit ec: ExecutionContext, bodyParsers: PlayBodyParsers)
     extends Controller {
 
+  /* to provide v8, remove legacy routes */
```

Comment:
Not sure I understand this comment. What legacy routes are removed? (I mean, the PR removes some, but this comment appears to be about the following code lines). Maybe it should be something like “to provide v8, add dataSet to task json?” Or maybe remove these comments altogether.

Reply:
> Or maybe remove these comments altogether.

Something like these comments were present previously (I think). Thus, I kept them to "order" the code a little :)

I can also remove them in case you prefer that :)